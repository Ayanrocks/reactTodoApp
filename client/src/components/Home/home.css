.form-card {
  position: absolute;
  top: 30%;
  left: 50%;
  transform: translate(-50%);
  padding: 25px;
  width: 40%;
  background-color: #fff;
  border: 1px;
  text-align: center;
  box-shadow: 0 1px 24px 5px rgba(0, 0, 0, 0.1);
}

.btn-custom {
  padding: 0.5rem 2rem;
  margin: 2rem 0;
  text-align: center;
}

.todoView {
  display: flex;
  flex-wrap: nowrap;
  width: 80%;
  margin: 0 auto;
}

label {
  font-size: 1.1rem;
  color: #999;
  margin-right: 1rem;
  display: inline-block;
}

p {
  display: inline-block;
  margin-left: 1rem;
  font-size: 1.5rem;
}

form {
  width: 100%;
}

input[type="text"] {
  width: 70%;
  color: #000;
  height: 2rem;
  padding: 0.4rem;
  font-size: 1.5rem;
  margin-left: 1em;
  display: inline-block;
}

select {
  margin-left: 0.7em;
  width: 70%;
  background-color: #fff;
  border: 1px solid rgb(187, 187, 187);
  font-size: 1.2rem;
  padding: 0.4rem;
  color: rgb(44, 44, 44);
}

button[type="submit"] {
  width: 85%;
  background-color: rgb(75, 228, 101);
  border: none;
  padding: 0.2rem;
  border: 1px solid rgba(56, 240, 102, 0.4);
  color: #fff;
}
