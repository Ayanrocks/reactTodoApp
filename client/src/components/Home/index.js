import React, { Component } from "react";
import TodoCard from "../todoCard";
import { connect } from "react-redux";
import { fetchTodo } from "../../actions";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: []
    };
  }
  async componentDidMount() {
    await this.props.fetchTodo();
    console.log(this.props);
    this.setState({ todos: this.props.todos[0] });
  }
  render() {
    return (
      <div>
        <div className="container mt-5">
          <div className="row">
            {this.state.todos &&
              this.state.todos.map((todo, i) => {
                return <TodoCard key={i} todo={todo} />;
              })}
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps({ todoReducer }) {
  const { todos } = todoReducer;
  return {
    todos
  };
}

export default connect(mapStateToProps, { fetchTodo })(Home);
