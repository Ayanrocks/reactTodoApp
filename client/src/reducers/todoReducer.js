import { FETCH_TODOS, FETCH_BUCKETS } from "../actions/types";

const initialState = {
  todos: [],
  buckets: []
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TODOS:
      return { ...state, todos: action.payload };
    case FETCH_BUCKETS:
      return { ...state, buckets: action.payload };
    default:
      return state;
  }
};
