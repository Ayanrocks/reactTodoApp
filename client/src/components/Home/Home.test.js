import React from "react";
import Home from "./index";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "../../reducers";

it("Home renders correctly", () => {
  const mockStore = createStore(reducers, {}, applyMiddleware(reduxThunk));
  const tree = renderer
    .create(
      <Provider store={mockStore}>
        <Home />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
