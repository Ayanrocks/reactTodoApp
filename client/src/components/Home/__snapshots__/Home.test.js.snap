// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row"
    />
  </div>
  <nav
    className="fab-container undefined"
  >
    <button
      className="fab-item undefined "
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f44336",
          "border": "none",
          "color": "#fafafa",
          "textDecoration": "none",
        }
      }
      tooltip="Create Todo"
    >
      <i
        className="fa fa-sticky-note"
      />
    </button>
    <button
      className="fab-item undefined "
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f44336",
          "border": "none",
          "color": "#fafafa",
          "textDecoration": "none",
        }
      }
      tooltip="Create Bucket"
    >
      <i
        className="fa fa-beer"
      />
    </button>
    <button
      className="fab-item undefined fab-rotate"
      style={
        Object {
          "backgroundColor": "#f44336",
          "border": "none",
          "color": "#fafafa",
          "textDecoration": "none",
        }
      }
      tooltip="Add"
    >
      <i
        className="fa fa-plus"
      />
    </button>
  </nav>
</div>
`;
